lib/exw3/contract.ex

```diff
@@ -335,9 +348,10 @@ defmodule ExW3.Contract do
         if Enum.member?(["latest", "earliest", "pending"], event_data[key]) do
           event_data[key]
         else
-          "0x" <>
-            (ExW3.Abi.encode_data("(uint256)", [event_data[key]])
-             |> Base.encode16(case: :lower))
+          event_data[key]
+          |> Integer.to_string(16)
+          |> String.downcase()
+          |> String.replace_prefix("", "0x")
```

Comment:
I had to change these lines because most RPC don't accept numbers with leading 0's like:
```
  Errors encountered in param 0: Invalid value "0x014f7c5b" supplied to : RpcFilterRequest/fromBlock: (QUANTITY | "earliest" | "latest" | "pending") | undefined/0: QUANTITY, Invalid value "0x0000000000000000000000000000000000000000000000000000000000000001" supplied to
: RpcFilterRequest/fromBlock: (QUANTITY | "earliest" | "latest" | "pending") | undefined/1: "earliest" | "latest" | "pending", Invalid value "0x0000000000000000000000000000000000000000000000000000000000000001" supplied to : RpcFilterRequest/toBlock: (QUANTITY | "earlie
st" | "latest" | "pending") | undefined/0: QUANTITY, Invalid value "0x0000000000000000000000000000000000000000000000000000000000000002" supplied to : RpcFilterRequest/toBlock: (QUANTITY | "earliest" | "latest" | "pending") | undefined/1: "earliest" | "latest" | "pendin
g"
```
Tested against hardhat native node & https://polygon-rpc.com & alchemy.

Reply:
I remember going about adding the 0x prefix a long time ago to fix other bugs. Maybe it isn't needed for logs.

@rupurt do you think this would break on other eth clients?